spring-cloud-bus/src/main/java/org/springframework/cloud/bus/event/RefreshListener.java

```diff
@@ -34,14 +35,25 @@
 
 	private ContextRefresher contextRefresher;
 
-	public RefreshListener(ContextRefresher contextRefresher) {
+	private ServiceMatcher serviceMatcher;
+
+	public RefreshListener(ContextRefresher contextRefresher,
+			ServiceMatcher serviceMatcher) {
 		this.contextRefresher = contextRefresher;
+		this.serviceMatcher = serviceMatcher;
 	}
 
 	@Override
 	public void onApplicationEvent(RefreshRemoteApplicationEvent event) {
-		Set<String> keys = this.contextRefresher.refresh();
-		log.info("Received remote refresh request. Keys refreshed " + keys);
+		log.info("Received remote refresh request.");
+		if (serviceMatcher.isForSelf(event)) {
```

Comment:
If we make this backward compatible the check should include, `serviceMatcher == null || ` though in practice it won't be since our auto config will use the new one.